predicators/datasets/vlm_input_data.py

```diff
@@ -340,3 +568,140 @@ def create_ground_atom_data_from_labeled_txt(
     low_level_trajs = _convert_ground_option_trajs_into_lowleveltrajs(
         option_trajs, goal_states_for_every_traj, train_tasks)
     return Dataset(low_level_trajs, ground_atoms_trajs)
+
+
+def create_ground_atom_data_from_img_trajs(
+        env: BaseEnv,
+        train_tasks: List[Task],
+        known_options: Set[ParameterizedOption],
+        vlm: Optional[VisionLanguageModel] = None) -> Dataset:
+    """Given a folder containing trajectories that have images of scenes for
+    each state, as well as options that transition between these states, output
+    a dataset."""
+    trajectories_folder_path = os.path.join(CFG.data_dir,
+                                            CFG.vlm_trajs_folder_name)
+    # First, run some checks on the folder name to make sure
+    # we're not accidentally loading the wrong one.
+    folder_name_components = CFG.vlm_trajs_folder_name.split('__')
+    assert folder_name_components[0] == CFG.env
+    assert folder_name_components[1] == "vlm_demos"
+    assert int(folder_name_components[2]) == CFG.seed
+    assert int(folder_name_components[3]) == CFG.num_train_tasks
+    num_trajs = len(os.listdir(trajectories_folder_path))
+    assert num_trajs == CFG.num_train_tasks
+    option_name_to_option = {opt.name: opt for opt in known_options}
+    image_option_trajs = []
+    all_task_objs = set()
+    for train_task_idx, path in enumerate(
+            sorted(Path(trajectories_folder_path).iterdir())):
+        assert path.is_dir()
+        state_folders = [f.path for f in os.scandir(path) if f.is_dir()]
+        num_states_in_traj = len(state_folders)
+        state_traj = []
+        for state_num in range(num_states_in_traj):
+            currimgs = []
+            curr_state_path = path.joinpath(str(state_num))
+            # NOTE: we assume all images are saved as jpg files.
+            img_files = glob.glob(str(curr_state_path) + "/*.jpg")
```

Comment:
is this order deterministic?

Reply:
no! wonderful catch; I'll make sure to unpack it in sorted order!